Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -1135,10 +1151,10 @@ public void ResetGatewayConnection()
             Disconnect();
 
             Log.Trace("InteractiveBrokersBrokerage.ResetGatewayConnection(): Stopping IB Gateway...");
-            InteractiveBrokersGatewayRunner.Stop();
+            _ibAutomater.Stop();
 
             Log.Trace("InteractiveBrokersBrokerage.ResetGatewayConnection(): Restarting IB Gateway...");
-            InteractiveBrokersGatewayRunner.Restart();
+            _ibAutomater.Start(false);
```

Comment:
Shouldn't we reset `_ibAutomaterInitializeEvent` before staring and waiting for it to be set? Also maybe call `CheckIbAutomaterErrors();`? Same doubt about it timing out.

Reply:
Good point! 
`CheckIbAutomaterErrors();` should be called here as well, I'll update and retest this.